fineract-provider/src/main/java/org/apache/fineract/portfolio/collateralmanagement/service/CollateralManagementWritePlatformServiceImpl.java

```diff
@@ -0,0 +1,185 @@
+import java.util.List;
+import org.apache.fineract.infrastructure.core.api.JsonCommand;
+import org.apache.fineract.infrastructure.core.data.ApiParameterError;
+import org.apache.fineract.infrastructure.core.data.CommandProcessingResult;
+import org.apache.fineract.infrastructure.core.data.CommandProcessingResultBuilder;
+import org.apache.fineract.infrastructure.core.data.DataValidatorBuilder;
+import org.apache.fineract.infrastructure.core.exception.PlatformApiDataValidationException;
+import org.apache.fineract.infrastructure.core.serialization.FromJsonHelper;
+import org.apache.fineract.organisation.monetary.domain.ApplicationCurrency;
+import org.apache.fineract.organisation.monetary.domain.ApplicationCurrencyRepository;
+import org.apache.fineract.portfolio.collateralmanagement.api.CollateralAPIConstants;
+import org.apache.fineract.portfolio.collateralmanagement.domain.ClientCollateralManagement;
+import org.apache.fineract.portfolio.collateralmanagement.domain.CollateralManagementDomain;
+import org.apache.fineract.portfolio.collateralmanagement.domain.CollateralManagementRepositoryWrapper;
+import org.apache.fineract.portfolio.collateralmanagement.exception.CollateralCannotBeDeletedException;
+import org.apache.fineract.portfolio.collateralmanagement.exception.CollateralNotFoundException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+public class CollateralManagementWritePlatformServiceImpl implements CollateralManagementWritePlatformService {
+
+    private final CollateralManagementRepositoryWrapper collateralManagementRepositoryWrapper;
+    private final ApplicationCurrencyRepository applicationCurrencyRepository;
+    private final FromJsonHelper fromApiJsonHelper;
+
+    @Autowired
+    public CollateralManagementWritePlatformServiceImpl(final CollateralManagementRepositoryWrapper collateralManagementRepositoryWrapper,
+            final ApplicationCurrencyRepository applicationCurrencyRepository, final FromJsonHelper fromApiJsonHelper) {
+        this.collateralManagementRepositoryWrapper = collateralManagementRepositoryWrapper;
+        this.applicationCurrencyRepository = applicationCurrencyRepository;
+        this.fromApiJsonHelper = fromApiJsonHelper;
+    }
+
+    @Transactional
+    @Override
+    public CommandProcessingResult createCollateral(final JsonCommand jsonCommand) {
+
+        /**
+         * TODO: Authenticate User
+         */
+        validateForCreation(jsonCommand);
+        final String currencyParamName = jsonCommand
+                .stringValueOfParameterNamed(CollateralAPIConstants.CollateralManagementJSONInputParams.CURRENCY.getValue());
+        final ApplicationCurrency applicationCurrency = this.applicationCurrencyRepository.findOneByCode(currencyParamName);
+
+        /**
+         * TODO: Check if application currency is empty.
+         */
```

Comment:
check this two todos

Reply:
These things are done. I'll remove comments.

Reply:
Fixed